### PR TITLE
OCM-8122 | ci: Added prefix/suffix to TF jobs

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/OCM-8122

### CODE REVIEW

**Comment by radtriste** [comment]
Depends on https://github.com/terraform-redhat/terraform-provider-rhcs/pull/666

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 1]
```diff
@@ -42,6 +42,8 @@ tests:
       CLUSTER_PROFILE: rosa-sts-ad
       QE_USAGE: rosa-sts-advanced-critical-high-f3
       REGION: ap-northeast-1
+      RHCS_CLUSTER_NAME_PREFIX: ci
+      RHCS_CLUSTER_NAME_SUFFIX: nightly
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:46]: It's not a nightly job. Why not put the frequency? like `f3`

Comment (radtriste) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:46]: yeah. wanted generic but putting the frequency works as well.
I mostly wanted to identify which resources are for PR and which are for nightly.
Will update

File: ci-operator/step-registry/rhcs/e2e/general-tests/rhcs-e2e-general-tests-ref.yaml [hunk 2]
```diff
@@ -39,3 +43,5 @@ ref:
     RHCS_URL should point to OCM url
     VERSION is set to latest by default, it will list the supported versions and then use the latest. Or you can set a uniqe value like 4.14.0
     REAGION will use the value set in profile configuration. Otherwise it will use the one in env settings
+    RHCS_CLUSTER_NAME_PREFIX will be used to identify the cluster and some other resources created
+    RHCS_CLUSTER_NAME_SUFFIX will be used to identify the cluster and some other resources created
```
Comment (yasun1) [inline ci-operator/step-registry/rhcs/e2e/general-tests/rhcs-e2e-general-tests-ref.yaml:47]: Why not write the description like this?
```
  - name: RHCS_CLUSTER_NAME_PREFIX
    default: ""
    documentation: It is used to identify the prefix of the cluster and some other resources.
```

File: ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh [hunk 3]
```diff
@@ -77,6 +77,12 @@ fi
 if [ ! -z "$RHCS_VERSION" ]; then
     export RHCS_VERSION=$RHCS_VERSION
 fi
+if [ ! -z "$RHCS_CLUSTER_NAME_PREFIX" ]; then
+    export RHCS_CLUSTER_NAME_PREFIX=$RHCS_CLUSTER_NAME_PREFIX
+fi
+if [ ! -z "$RHCS_CLUSTER_NAME_SUFFIX" ]; then
+    export RHCS_CLUSTER_NAME_SUFFIX=$RHCS_CLUSTER_NAME_SUFFIX
+fi
```
Comment (yasun1) [inline ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh:85]: Actually, we do not need to `export` the pod's environment variable again unless we want to do more around it.

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 4]
```diff
@@ -392,6 +436,8 @@ tests:
       CLUSTER_PROFILE: rosa-hcp-pl
       QE_USAGE: rosa-hcp-private-optional-presubmit
       REGION: us-west-2
+      RHCS_CLUSTER_NAME_PREFIX: ci
```
Comment (yasun1) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:439]: If `RHCS_CLUSTER_NAME_PREFIX` is always `ci` why not hardcode it in the step to reduce variable use? Besides, for you have defined `QE_USAGE`, maybe `RHCS_CLUSTER_NAME_SUFFIX` also can be defined in the step by it.

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 5]
```diff
@@ -392,6 +436,8 @@ tests:
...
+      RHCS_CLUSTER_NAME_SUFFIX: pr
```
Comment (yasun1) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:440]: If `RHCS_CLUSTER_NAME_PREFIX` is always ci why not hardcode it in the step to reduce variable use? Besides, for there is `QE_USAGE`, maybe `RHCS_CLUSTER_NAME_SUFFIX` also can be defined in the step by dealing with it.

File: ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh (hunk 3 above)
Comment (radtriste) [reply ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh:85]: I was not sure about that

**Comment by radtriste** [comment]
/pj-rehearsal pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-hcp-advanced-critical-high-presubmit

**Comment by radtriste** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-hcp-advanced-critical-high-presubmit

**Comment by radtriste** [comment]
`CLUSTER_NAME = ci-rhcs-hcp-ad-icl-pr`, as expected

**Pull request merged**